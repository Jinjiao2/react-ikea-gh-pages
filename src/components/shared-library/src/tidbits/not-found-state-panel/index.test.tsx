/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import React from 'react'

import theme from '@tidbits/react-tidbits/theme'

import { NotFoundStatePanel } from './index'

describe('State panel component', () => {
  test('renders correctly', async () => {
    const tree = render(
      <ThemeProvider theme={theme}>
        <NotFoundStatePanel />
      </ThemeProvider>,
    )
    expect(tree).toMatchSnapshot()
  })

  test('should return the element by text', async () => {
    render(
      <ThemeProvider theme={theme}>
        <NotFoundStatePanel />
      </ThemeProvider>,
    )
    expect(
      screen.getByText("Don't panic. Something went wrong."),
    ).toBeInTheDocument()
  })
})
