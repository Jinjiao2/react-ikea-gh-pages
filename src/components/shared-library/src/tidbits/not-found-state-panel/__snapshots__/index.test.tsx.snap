// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`State panel component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Box-sc-18eybku-0 kGhaHC"
        display="flex"
      >
        <div
          class="Box-sc-18eybku-0 bCznPx"
        >
          <h3
            class="Box-sc-18eybku-0 Text__BaseText-sc-4nqr4q-0 Text-sc-4nqr4q-1 kwciFV cHQCpL gClPCo"
            color="title"
          >
            Don't panic. Something went wrong.
          </h3>
          <h5
            class="Box-sc-18eybku-0 Text__BaseText-sc-4nqr4q-0 Text-sc-4nqr4q-1 jkGwVr cHQCpL gClPCo"
            color="title"
          >
            Browser doesn't like the page you were trying to load.
          </h5>
          <p
            class="Box-sc-18eybku-0 Text__BaseText-sc-4nqr4q-0 Text-sc-4nqr4q-1 iekmwW cHQCpL cpYCss"
            color="label"
          >
            Please check the url and try again.
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Box-sc-18eybku-0 kGhaHC"
      display="flex"
    >
      <div
        class="Box-sc-18eybku-0 bCznPx"
      >
        <h3
          class="Box-sc-18eybku-0 Text__BaseText-sc-4nqr4q-0 Text-sc-4nqr4q-1 kwciFV cHQCpL gClPCo"
          color="title"
        >
          Don't panic. Something went wrong.
        </h3>
        <h5
          class="Box-sc-18eybku-0 Text__BaseText-sc-4nqr4q-0 Text-sc-4nqr4q-1 jkGwVr cHQCpL gClPCo"
          color="title"
        >
          Browser doesn't like the page you were trying to load.
        </h5>
        <p
          class="Box-sc-18eybku-0 Text__BaseText-sc-4nqr4q-0 Text-sc-4nqr4q-1 iekmwW cHQCpL cpYCss"
          color="label"
        >
          Please check the url and try again.
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
